### PR TITLE
Support Terraform deployment files

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR introduces new Terraform-based deployment service definitions. Deployment definitions will enabled for both data stream and package (reuse). 

## Why is it important?

We're moving on with service deployer for AWS, GCP, etc.

## Checklist

- [x] I have added test packages to [`code/go/internal/validator/test/packages`](https://github.com/elastic/package-spec/tree/master/code/go/internal/validator/test/packages) that prove my change is effective.
- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

## Related issues

- https://github.com/elastic/elastic-package/issues/89

### CODE REVIEW

File: versions/1/data_stream/_dev/spec.yml
Comment (ycombinator) [inline versions/1/data_stream/_dev/spec.yml:13]: Is this what we want? It implies that there can be data stream level deployment definitions for Docker Compose-based deployments as well, not just Terraform-based deployments. Having such consistency is really nice but I want to make sure it's something that's actually going to be supported by `elastic-package` as well. Otherwise the spec is a bit misleading.

Comment (mtojek) [reply versions/1/data_stream/_dev/spec.yml:13]: Yes, I modified the `elastic-package` here: https://github.com/elastic/elastic-package/pull/228 . 

As you noticed, It means that we will support deploy definitions on the data stream level. Actually we may leverage from this change, as currently developers build a single Docker image for multiple data streams. See: https://github.com/elastic/integrations/tree/master/packages/zeek/_dev/deploy/docker

If you prefer to support `tf` only, I can adjust the spec.

**Comment by mtojek** [comment]
I will solve all conflicts (`statik.go` and `changelog.yml`) once everything else is fine.

File: versions/1/data_stream/_dev/spec.yml
Comment (ycombinator) [reply versions/1/data_stream/_dev/spec.yml:13]: Ah nice, I hadn't studied https://github.com/elastic/elastic-package/pull/228 in depth before, sorry! Given what you have there, what you have here is good as-is. Thanks!

**Review by ycombinator — APPROVED** [review]
LGTM once merge conflicts are addressed and CI is green.

**Pull request merged**
